modules/core/04-channel/keeper/upgrade.go

```diff
@@ -237,6 +237,10 @@ func (k Keeper) ChanUpgradeAck(
 		return errorsmod.Wrap(err, "failed to retrieve connection using the channel connection hops")
 	}
 
+	if connection.GetState() != int32(connectiontypes.OPEN) {
```

Comment:
this check is already made in `startFlushUpgradeHandshake`, but I don't mind to do it here also - I guess it saves gas on the upgrade store lookup below before calling into `startFlushUpgradeHandshake`.

Reply:
Yea, not strictly necessary, but I guess extra defensive. Might be nice to eventually have an API that only returns open connections and returns an error otherwise

Reply:
https://github.com/cosmos/ibc-go/issues/3466